nx/lib/nx/defn/kernel.ex

```diff
@@ -946,6 +946,11 @@ defmodule Nx.Defn.Kernel do
             "got: #{Macro.to_string(other)}"
   end
 
+  def complex(real, imag \\ 0) do
+    imag_component = Nx.multiply(Nx.tensor(imag), Nx.tensor(Complex.new(0, 1)))
+    Nx.add(real, imag_component)
```

Comment:
Actually, perhaps we could just move this do Nx instead of keeping it here in the kernel.

It would be the inverse operation of `real` and `imag` such that: `t == complex(real(t), imag(t))`

WDYT?

Reply:
I am afraid this version would be an expensive operation because we need to get the real part, the imaginary part, multiply imaginary part by the constant `i`, and add them together. So I would go with a number only complex/2 Function, Which is also What python offers, and the `i()` constant in Nx.Constants (which people can import).

Reply:
It's not that we need to get the real and imaginary parts, but that we have them and want to construct a complex tensor from them. Pytorch allows for such cases:

```python
In [1]: import torch

In [2]: torch.complex(torch.tensor([1.0, 2.0, 3.0]), torch.tensor([4.0, 5.0, 6.0]))
Out[2]: tensor([1.+4.j, 2.+5.j, 3.+6.j])
```

We could either have that as an optional callback with this implementation as the default, or as a callback in itself.

Reply:
Oh, I see. So what happens in this case:

    torch.complex(torch.tensor([1.0, 2.0, 3.0]), torch.tensor([1.+4.j, 2.+5.j, 3.+6.j]))

?

Reply:
torch only accepts floating tensors as torch.complex inputs:

```python
In [2]: import torch

In [3]: real = torch.tensor([1.0, 2.0])

In [4]: complex = torch.complex(real, torch.tensor([3.0, 4.0]))

In [5]: complex
Out[5]: tensor([1.+3.j, 2.+4.j])

In [6]: torch.complex(real, compleX)
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-6-191e657a427d> in <module>
----> 1 torch.complex(real, compleX)

NameError: name 'compleX' is not defined

In [7]: torch.complex(real, complex)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-7-949ebd635540> in <module>
----> 1 torch.complex(real, complex)

RuntimeError: Expected both inputs to be Float or Double tensors but got Float and ComplexFloat

In [8]: torch.complex(real, torch.tensor([3, 4]))
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-8-321f23cbb08d> in <module>
----> 1 torch.complex(real, torch.tensor([3, 4]))

RuntimeError: Expected both inputs to be Float or Double tensors but got Float and Long
```

Reply:
Ok, perfect. Let's go with Nx.complex/2 then. It can accept anything except complex. We still need `Nx.Constants.i()` though.